packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp

```diff
@@ -0,0 +1,131 @@
+  @scenarioDoc(
+    """
+  Expected input body:
+  ```json
+  {doc}
+  ```
+  """,
+    {
+      doc: TDoc,
+    }
+  )
+  @put
+  @doc("Put operation")
+  put(@body @doc("body") body: TModel): void;
+}
+
+// ********************************************** Record<unknown> **********************************************
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model extends from Record<unknown> type.")
+model ExtendsUnknownAddtionalProperties extends Record<unknown> {
+  @doc("The name property")
+  name: string;
+}
+
+@route("/extendsRecordUnknown")
+@operationGroup
+interface ExtendsUnknown
+  extends ModelOperations<
+      ExtendsUnknownAddtionalProperties,
+      "{'name': 'ExtendsUnknownAddtionalProperties', 'prop1': 32, 'prop2': true, 'prop3': 'abc'}"
+    > {}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model is from Record<unknown> type.")
+model IsUnknownAddtionalProperties is Record<unknown> {
+  @doc("The name property")
+  name: string;
+}
+
+@route("/isRecordUnknown")
+@operationGroup
+interface IsUnknown
+  extends ModelOperations<
+      IsUnknownAddtionalProperties,
+      "{'name': 'IsUnknownAddtionalProperties', 'prop1': 32, 'prop2': true, 'prop3': 'abc'}"
+    > {}
+
+// ********************************************** Record<string> **********************************************
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model extends from Record<string> type.")
+model ExtendsStringAddtionalProperties extends Record<string> {
+  @doc("The name property")
+  name: string;
+}
+
+@route("/extendsRecordString")
+@operationGroup
+interface ExtendsString
+  extends ModelOperations<
+      ExtendsStringAddtionalProperties,
+      "{'name': 'ExtendsStringAddtionalProperties', 'prop': 'abc'}"
+    > {}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model is from Record<string> type.")
+model IsStringAddtionalProperties is Record<string> {
+  @doc("The name property")
+  name: string;
+}
+
+@route("/isRecordstring")
+@operationGroup
+interface IsString
+  extends ModelOperations<IsStringAddtionalProperties, "{'name': 'IsStringAddtionalProperties', 'prop': 'abc'}"> {}
+
+// ********************************************** Record<float32> **********************************************
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model extends from Record<float32> type.")
+model ExtendsFloatAddtionalProperties extends Record<float32> {
+  @doc("The id property")
+  id: float32;
+}
+
+@route("/extendsRecordFloat")
+@operationGroup
+interface ExtendsFloat extends ModelOperations<ExtendsFloatAddtionalProperties, "{'id': 42.42, 'prop': 42.42}"> {}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "For testing"
+@doc("The model is from Record<float32> type.")
+model IsFloatAddtionalProperties is Record<float32> {
```

Comment:
Can you add test cases for extending from Record of Model. as well as model arrays ? we have a customer asking for this https://github.com/Azure/autorest.typescript/issues/2033. Thanks.

Reply:
Added testcase for Model and Model[]. @qiaozha @chunyu3 Please help review